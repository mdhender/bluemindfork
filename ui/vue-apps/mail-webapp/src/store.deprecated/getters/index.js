export * from "./areAllMessagesSelected";
export * from "./areAllSelectedMessagesFlagged";
export * from "./areAllSelectedMessagesRead";
export * from "./areAllSelectedMessagesUnflagged";
export * from "./areAllSelectedMessagesUnread";
export * from "./areRemoteImagesUnblocked";
export * from "./isMessageSelected";
export * from "./lastRecipients";
export * from "./nextMessageKey";
export * from "./status";
