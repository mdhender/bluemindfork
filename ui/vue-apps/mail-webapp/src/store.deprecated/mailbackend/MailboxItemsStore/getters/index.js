export * from "./count";
export * from "./indexOf";
