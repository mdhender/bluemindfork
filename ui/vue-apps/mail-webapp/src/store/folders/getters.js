import { FOLDER_BY_PATH, FOLDER_HAS_CHILDREN } from "~getters";

export default {
    [FOLDER_BY_PATH]: state => path => Object.values(state).find(folder => folder.path === path),
    [FOLDER_HAS_CHILDREN]: state => key => !!Object.values(state).find(folder => folder.parent === key)
};
