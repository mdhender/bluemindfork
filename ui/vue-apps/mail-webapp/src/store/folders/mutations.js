import { Flag } from "@bluemind/email";
import Vue from "vue";
import { RENAME_FOLDER } from "~actions";
import {
    ADD_FLAG,
    ADD_FOLDER,
    DELETE_FLAG,
    MOVE_MESSAGES,
    REMOVE_FOLDER,
    REMOVE_MESSAGES,
    SET_ACTIVE_FOLDER,
    SET_FOLDER_EXPANDED,
    SET_MAILBOX_FOLDERS,
    SET_UNREAD_COUNT
} from "~mutations";

export default {
    [ADD_FOLDER]: (state, { key, ...folder }) => {
        Vue.set(state, key, { ...folder, key });
    },
    [SET_MAILBOX_FOLDERS]: (state, { folders }) => {
        folders.forEach(folder => {
            Vue.set(state, folder.key, folder);
        });
    },
    [RENAME_FOLDER]: (state, { key, name, path }) => {
        state[key].name = name;
        state[key].path = path;
    },
    [REMOVE_FOLDER]: (state, { key }) => {
        Vue.delete(state, key);
    },
    [SET_UNREAD_COUNT]: (state, { key, unread }) => {
        state[key].unread = unread;
    },
    [SET_FOLDER_EXPANDED]: (state, { key, expanded }) => {
        state[key].expanded = expanded;
    },
    // Hooks
    [ADD_FLAG]: (state, { messages, flag }) => {
        if (flag === Flag.SEEN) {
            messages.forEach(({ folderRef: { key } }) => state[key] && state[key].unread--);
        }
    },
    [DELETE_FLAG]: (state, { messages, flag }) => {
        if (flag === Flag.SEEN) {
            messages.forEach(({ folderRef: { key } }) => state[key] && state[key].unread++);
        }
    },
    [SET_ACTIVE_FOLDER]: (state, folder) => {
        let parent = folder.parent && state[folder.parent];
        while (parent) {
            parent.expanded = true;
            parent = parent.parent && state[parent.parent];
        }
    }
};
