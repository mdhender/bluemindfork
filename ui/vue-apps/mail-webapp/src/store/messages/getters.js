function activeFilterToFunction(ACTIVE_FLAG = {}) {
    return message => {
        if (message.data && message.data.flags) {
            return Object.entries(ACTIVE_FLAG).reduce(
                (match, [key, value]) => match && message.data.flags[key] === value,
                true
            );
        }
    };
}

export const ACTIVE_MESSAGES = "ACTIVE_MESSAGES";
const activeMessages = function activeMessages(state, { ACTIVE_FOLDER }) {
    return Object.entries(state)
        .map(([, message]) => message)
        .filter(message => message.folder === ACTIVE_FOLDER);
};

export const FILTERED_MESSAGES_BY_FLAG = "FILTERED_MESSAGES_BY_FLAG";
const filteredMessagesByFlag = function filteredMessages(state, { ACTIVE_FLAG }) {
    const activeFilter = activeFilterToFunction(ACTIVE_FLAG);
    return Object.entries(state)
        .map(([, message]) => message)
        .filter(activeFilter);
};

export default {
    [ACTIVE_MESSAGES]: activeMessages,
    [FILTERED_MESSAGES_BY_FLAG]: filteredMessagesByFlag
};

//TODO: voici les getters qui sont utilisées ci et là
// const count = state => {};
// const getMessagesByKey = (state, { messages }) => {};
// const getPartContent = state => {};
// const messages = state => {};
