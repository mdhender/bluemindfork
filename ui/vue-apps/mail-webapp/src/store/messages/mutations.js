import Vue from "vue";
import { MessageStatus } from "../../model/message";
import {
    ADD_ATTACHMENT,
    ADD_FLAG,
    ADD_MESSAGES,
    DELETE_FLAG,
    REMOVE_ATTACHMENT,
    REMOVE_MESSAGES,
    MOVE_MESSAGES,
    SET_ATTACHMENT_ADDRESS,
    SET_ATTACHMENT_ENCODING,
    SET_ATTACHMENT_PROGRESS,
    SET_ATTACHMENT_STATUS,
    SET_MESSAGES_STATUS,
    SET_MESSAGE_BCC,
    SET_MESSAGE_CC,
    SET_MESSAGE_COMPOSING,
    SET_MESSAGE_DATE,
    SET_MESSAGE_HAS_ATTACHMENT,
    SET_MESSAGE_HEADERS,
    SET_MESSAGE_INTERNAL_ID,
    SET_MESSAGE_IMAP_UID,
    SET_MESSAGE_PREVIEW,
    SET_MESSAGE_LIST,
    SET_MESSAGE_SUBJECT,
    SET_MESSAGE_TO,
    SET_UNREAD_COUNT
} from "~mutations";
import { Flag } from "@bluemind/email";

export default {
    [ADD_MESSAGES]: (state, messages) => {
        messages.forEach(message => {
            Vue.set(state, message.key, message);
        });
    },
    [ADD_FLAG]: (state, { messages, flag }) => {
        messages.forEach(({ key }) => state[key].flags.push(flag));
    },
    [DELETE_FLAG]: (state, { messages, flag }) => {
        messages.forEach(({ key }) => (state[key].flags = state[key].flags.filter(f => f !== flag)));
    },
    [REMOVE_MESSAGES]: removeMessages,
    [MOVE_MESSAGES]: (state, { messages }) => removeMessages(state, messages),
    [SET_UNREAD_COUNT]: (state, { key, unread }) => {
        if (unread === 0) {
            //FIXME: This cannot be rolled back...
            Object.values(state).forEach(({ status, folderRef, flags }) => {
                if (status === MessageStatus.LOADED && folderRef.key === key && !flags.includes(Flag.SEEN)) {
                    flags.push(Flag.SEEN);
                }
            });
        }
    },
    [SET_MESSAGE_PREVIEW]: (state, { key, preview }) => {
        state[key].preview = preview;
    },
    [SET_MESSAGES_STATUS]: (state, messages) => {
        messages.forEach(m => (state[m.key].status = m.status));
    },
    [SET_MESSAGE_LIST]: (state, messages) => {
        messages.filter(message => !state[message.key]).forEach(message => Vue.set(state, message.key, message));
    },
    [SET_MESSAGE_COMPOSING]: (state, { messageKey, composing }) => {
        state[messageKey].composing = composing;
    },
    [SET_MESSAGE_SUBJECT]: (state, { messageKey, subject }) => {
        state[messageKey].subject = subject;
    },
    [SET_MESSAGE_DATE]: (state, { messageKey, date }) => {
        state[messageKey].date = date;
    },
    [SET_MESSAGE_HAS_ATTACHMENT]: (state, { key, hasAttachment }) => {
        state[key].hasAttachment = hasAttachment;
    },
    [SET_MESSAGE_HEADERS]: (state, { messageKey, headers }) => {
        state[messageKey].headers = headers;
    },
    [SET_MESSAGE_TO]: (state, { messageKey, to }) => {
        state[messageKey].to = to;
    },
    [SET_MESSAGE_CC]: (state, { messageKey, cc }) => {
        state[messageKey].cc = cc;
    },
    [SET_MESSAGE_BCC]: (state, { messageKey, bcc }) => {
        state[messageKey].bcc = bcc;
    },
    [SET_MESSAGE_INTERNAL_ID]: (state, { key, internalId }) => {
        state[key].remoteRef.internalId = internalId;
    },
    [SET_MESSAGE_IMAP_UID]: (state, { key, imapUid }) => {
        state[key].remoteRef.imapUid = imapUid;
    },
    [ADD_ATTACHMENT]: (state, { messageKey, attachment }) => {
        state[messageKey].attachments.push(attachment);
    },
    [REMOVE_ATTACHMENT]: (state, { messageKey, address }) => {
        const attachments = state[messageKey].attachments;
        const index = attachments.findIndex(a => a.address === address);
        attachments.splice(index, 1);
    },
    [SET_ATTACHMENT_ADDRESS]: (state, { messageKey, oldAddress, address }) => {
        const attachment = state[messageKey].attachments.find(a => a.address === oldAddress);
        if (attachment) {
            attachment.address = address;
        }
    },
    [SET_ATTACHMENT_STATUS]: (state, { messageKey, address, status }) => {
        const attachment = state[messageKey].attachments.find(a => a.address === address);
        attachment.status = status;
    },
    [SET_ATTACHMENT_ENCODING]: (state, { messageKey, address, charset, encoding }) => {
        const attachment = state[messageKey].attachments.find(a => a.address === address);
        attachment.charset = charset;
        attachment.encoding = encoding;
    },
    [SET_ATTACHMENT_PROGRESS]: (state, { messageKey, address, loaded, total }) => {
        const attachment = state[messageKey].attachments.find(a => a.address === address);
        attachment.progress = { loaded, total };
    }
};
function removeMessages(state, messages) {
    messages.forEach(({ key }) => {
        Vue.delete(state, key);
    });
}
