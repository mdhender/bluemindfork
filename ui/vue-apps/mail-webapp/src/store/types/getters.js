// root
export const ACTIVE_MESSAGE = "ACTIVE_MESSAGE";
export const ALL_CONVERSATIONS_ARE_SELECTED = "ALL_CONVERSATIONS_ARE_SELECTED";
export const ALL_SELECTED_CONVERSATIONS_ARE_FLAGGED = "ALL_SELECTED_CONVERSATIONS_ARE_FLAGGED";
export const ALL_SELECTED_CONVERSATIONS_ARE_READ = "ALL_SELECTED_CONVERSATIONS_ARE_READ";
export const ALL_SELECTED_CONVERSATIONS_ARE_UNFLAGGED = "ALL_SELECTED_CONVERSATIONS_ARE_UNFLAGGED";
export const ALL_SELECTED_CONVERSATIONS_ARE_UNREAD = "ALL_SELECTED_CONVERSATIONS_ARE_UNREAD";
export const ALL_SELECTED_CONVERSATIONS_ARE_WRITABLE = "ALL_SELECTED_CONVERSATIONS_ARE_WRITABLE";
export const CONVERSATIONS_ACTIVATED = "CONVERSATIONS_ACTIVATED";
export const CURRENT_MAILBOX = "CURRENT_MAILBOX";
export const FILTERED_GROUP_MAILBOX_RESULTS = "FILTERED_GROUP_MAILBOX_RESULTS";
export const FILTERED_MAILSHARE_RESULTS = "FILTERED_MAILSHARE_RESULTS";
export const FILTERED_USER_RESULTS = "FILTERED_USER_RESULTS";
export const FOLDER_LIST_IS_EMPTY = "FOLDER_LIST_IS_EMPTY";
export const GROUP_MAILBOX_FOLDERS = "GROUP_MAILBOX_FOLDERS";
export const GROUP_MAILBOX_ROOT_FOLDERS = "GROUP_MAILBOX_ROOT_FOLDERS";
export const IS_ACTIVE_MESSAGE = "IS_ACTIVE_MESSAGE";
export const IS_CURRENT_CONVERSATION = "IS_CURRENT_CONVERSATION";
export const MAILBOX_FOLDERS = "MAILBOX_FOLDERS";
export const MAILBOX_ROOT_FOLDERS = "MAILBOX_ROOT_FOLDERS";
export const MAILBOX_INBOX = "MAILBOX_INBOX";
export const MAILBOX_JUNK = "MAILBOX_JUNK";
export const MAILBOX_SENT = "MAILBOX_SENT";
export const MAILBOX_TRASH = "MAILBOX_TRASH";
export const MAILSHARE_FOLDERS = "MAILSHARE_FOLDERS";
export const MAILSHARE_ROOT_FOLDERS = "MAILSHARE_ROOT_FOLDERS";
export const MY_DRAFTS = "MY_DRAFTS";
export const MY_INBOX = "MY_INBOX";
export const MY_JUNK = "MY_JUNK";
export const MY_MAILBOX_FOLDERS = "MY_MAILBOX_FOLDERS";
export const MY_MAILBOX_IS_READY = "MY_MAILBOX_IS_READY";
export const MY_MAILBOX_ROOT_FOLDERS = "MY_MAILBOX_ROOT_FOLDERS";
export const MY_OUTBOX = "MY_OUTBOX";
export const MY_SENT = "MY_SENT";
export const MY_TEMPLATES = "MY_TEMPLATES";
export const MY_TRASH = "MY_TRASH";
export const NEXT_CONVERSATION = "NEXT_CONVERSATION";
export const SELECTION = "SELECTION";
export const SELECTION_FLAGS = "SELECTION_FLAGS";
export const ADDRESS_AUTOCOMPLETE = "ADDRESS_AUTOCOMPLETE";

// mailboxes
export const GROUP_MAILBOXES = "GROUP_MAILBOXES";
export const GROUP_MAILBOX_KEYS = "GROUP_MAILBOX_KEYS";
export const MAILBOX_BY_NAME = "MAILBOX_BY_NAME";
export const MAILBOXES = "MAILBOXES";
export const MAILBOXES_ARE_LOADED = "MAILBOXES_ARE_LOADED";
export const MAILSHARE_KEYS = "MAILSHARE_KEYS";
export const MAILSHARES = "MAILSHARES";
export const MY_MAILBOX = "MY_MAILBOX";
export const MY_MAILBOX_KEY = "MY_MAILBOX_KEY";
export const USER_MAILBOXES = "USER_MAILBOXES";

// folders
export const FOLDER_BY_PATH = "FOLDER_BY_PATH";
export const FOLDER_GET_CHILDREN = "FOLDER_GET_CHILDREN";
export const FOLDER_GET_DESCENDANTS = "FOLDER_GET_DESCENDANTS";
export const FOLDER_HAS_CHILDREN = "FOLDER_HAS_CHILDREN";
export const FOLDERS = "FOLDERS";
export const FOLDERS_BY_PATH = "FOLDERS_BY_PATH";
export const IS_DESCENDANT = "IS_DESCENDANT";

// messages
export const MESSAGE_IS_LOADED = "MESSAGE_IS_LOADED";
export const MESSAGE_IS_LOADING = "MESSAGE_IS_LOADING";

// selection
export const SELECTION_IS_EMPTY = "SELECTION_IS_EMPTY";
export const ONE_CONVERSATION_SELECTED = "ONE_CONVERSATION_SELECTED";
export const SEVERAL_CONVERSATIONS_SELECTED = "SEVERAL_CONVERSATIONS_SELECTED";
export const CONVERSATION_IS_SELECTED = "CONVERSATION_IS_SELECTED";
export const SELECTION_KEYS = "SELECTION_KEYS";

// conversationList
export const CONVERSATION_LIST_ALL_KEYS = "CONVERSATION_LIST_ALL_KEYS";
export const CONVERSATION_LIST_COUNT = "CONVERSATION_LIST_COUNT";
export const CONVERSATION_LIST_FILTERED = "CONVERSATION_LIST_FILTERED";
export const CONVERSATION_LIST_FLAGGED_FILTER_ENABLED = "CONVERSATION_LIST_FLAGGED_FILTER_ENABLED";
export const CONVERSATION_LIST_HAS_NEXT = "CONVERSATION_LIST_HAS_NEXT";
export const CONVERSATION_LIST_IS_LOADING = "CONVERSATION_LIST_IS_LOADING";
export const CONVERSATION_LIST_IS_REJECTED = "CONVERSATION_LIST_IS_REJECTED";
export const CONVERSATION_LIST_IS_RESOLVED = "CONVERSATION_LIST_IS_RESOLVED";
export const CONVERSATION_LIST_KEYS = "CONVERSATION_LIST_KEYS";
export const CONVERSATION_LIST_TOTAL_PAGES = "CONVERSATION_LIST_TOTAL_PAGES";
export const CONVERSATION_LIST_UNREAD_FILTER_ENABLED = "CONVERSATION_LIST_UNREAD_FILTER_ENABLED";

// search
export const CONVERSATION_LIST_IS_FILTERED = "CONVERSATION_LIST_IS_FILTERED";
export const HAS_PATTERN = "HAS_PATTERN";
export const IS_TYPING_IN_SEARCH = "IS_TYPING_IN_SEARCH";

// conversations
export const CONVERSATION_IS_LOADED = "CONVERSATION_IS_LOADED";
export const CONVERSATION_MESSAGE_BY_KEY = "CONVERSATION_MESSAGE_BY_KEY";
export const CONVERSATION_METADATA = "CONVERSATION_METADATA";
export const CURRENT_CONVERSATION_METADATA = "CURRENT_CONVERSATION_METADATA";

// folderList
export const FOLDER_LIST_IS_FILTERED = "FOLDER_LIST_IS_FILTERED";
export const FOLDER_LIST_IS_LOADING = "FOLDER_LIST_IS_LOADING";
export const FOLDER_LIST_LIMIT_FOR_GROUP_MAILBOX = "FOLDER_LIST_LIMIT_FOR_GROUP_MAILBOX";
export const FOLDER_LIST_LIMIT_FOR_MAILSHARE = "FOLDER_LIST_LIMIT_FOR_MAILSHARE";
export const FOLDER_LIST_LIMIT_FOR_USER = "FOLDER_LIST_LIMIT_FOR_USER";

// messageCompose
export const IS_SENDER_SHOWN = "IS_SENDER_SHOWN";
export const GET_DRAFT_CONTENT = "GET_DRAFT_CONTENT";
