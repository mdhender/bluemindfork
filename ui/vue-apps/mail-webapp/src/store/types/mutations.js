// popup
export const IS_POPUP = "IS_POPUP";

// activeMessage
export const RESET_ACTIVE_MESSAGE = "RESET_ACTIVE_MESSAGE";
export const SET_ACTIVE_MESSAGE = "SET_ACTIVE_MESSAGE";

// partsData
export const RESET_PARTS_DATA = "RESET_PARTS_DATA";
export const SET_PART_DATA = "SET_PART_DATA";
export const SET_QUOTE_NODES = "SET_QUOTE_NODES";

// mailboxes
export const ADD_MAILBOXES = "ADD_MAILBOXES";

// folders
export const ADD_FOLDER = "ADD_FOLDER";
export const REMOVE_FOLDER = "REMOVE_FOLDER";
export const SET_ACTIVE_FOLDER = "SET_ACTIVE_FOLDER";
export const SET_UNREAD_COUNT = "SET_UNREAD_COUNT";
export const SET_FOLDER_EXPANDED = "SET_FOLDER_EXPANDED";
export const UPDATE_FOLDER = "UPDATE_FOLDER";
export const UPDATE_PATHS = "UPDATE_PATHS";

// message
export const ADD_ATTACHMENT = "ADD_ATTACHMENT";
export const ADD_FLAG = "ADD_FLAG";
export const ADD_MESSAGES = "ADD_MESSAGES";
export const DELETE_FLAG = "DELETE_FLAG";
export const MOVE_MESSAGES = "MOVE_MESSAGES";
export const REMOVE_ATTACHMENT = "REMOVE_ATTACHMENT";
export const REMOVE_MESSAGES = "REMOVE_MESSAGES";
export const SET_ATTACHMENT_ADDRESS = "SET_ATTACHMENT_ADDRESS";
export const SET_ATTACHMENT_ENCODING = "SET_ATTACHMENT_ENCODING";
export const SET_ATTACHMENT_HEADERS = "SET_ATTACHMENT_HEADERS";
export const SET_ATTACHMENT_PROGRESS = "SET_ATTACHMENT_PROGRESS";
export const SET_ATTACHMENT_STATUS = "SET_ATTACHMENT_STATUS";
export const SET_ATTACHMENTS = "SET_ATTACHMENTS";
export const SET_MESSAGE_BCC = "SET_MESSAGE_BCC";
export const SET_MESSAGE_CC = "SET_MESSAGE_CC";
export const SET_MESSAGE_COMPOSING = "SET_MESSAGE_COMPOSING";
export const SET_MESSAGE_DATE = "SET_MESSAGE_DATE";
export const SET_MESSAGE_FROM = "SET_MESSAGE_FROM";
export const SET_MESSAGE_HAS_ATTACHMENT = "SET_MESSAGE_HAS_ATTACHMENT";
export const SET_MESSAGE_HEADERS = "SET_MESSAGE_HEADERS";
export const SET_MESSAGE_INTERNAL_ID = "SET_MESSAGE_INTERNAL_ID";
export const SET_MESSAGE_INLINE_PARTS_BY_CAPABILITIES = "SET_MESSAGE_INLINE_PARTS_BY_CAPABILITIES";
export const SET_MESSAGE_IMAP_UID = "SET_MESSAGE_IMAP_UID";
export const SET_MESSAGES_LOADING_STATUS = "SET_MESSAGES_LOADING_STATUS";
export const SET_MESSAGE_PREVIEW = "SET_MESSAGE_PREVIEW";
export const SET_MESSAGE_SIZE = "SET_MESSAGE_SIZE";
export const SET_MESSAGES_STATUS = "SET_MESSAGES_STATUS";
export const SET_MESSAGE_SUBJECT = "SET_MESSAGE_SUBJECT";
export const SET_MESSAGE_TMP_ADDRESSES = "SET_MESSAGE_TMP_ADDRESSES";
export const SET_MESSAGE_TO = "SET_MESSAGE_TO";

//conversationList
export const RESET_CONVERSATION_LIST_PAGE = "RESET_CONVERSATION_LIST_PAGE";
export const SET_CONVERSATION_LIST = "SET_CONVERSATION_LIST";
export const SET_CONVERSATION_LIST_FILTER = "SET_CONVERSATION_LIST_FILTER";
export const SET_CONVERSATION_LIST_STATUS = "SET_CONVERSATION_LIST_STATUS";
export const SET_CONVERSATION_LIST_PAGE = "SET_CONVERSATION_LIST_PAGE";

//search
export const SET_SEARCH_FOLDER = "SET_SEARCH_FOLDER";
export const SET_SEARCH_PATTERN = "SET_SEARCH_PATTERN";

// messageCompose
export const MAX_MESSAGE_SIZE_EXCEEDED = "MAX_MESSAGE_SIZE_EXCEEDED";
export const RESET_COMPOSER = "RESET_COMPOSER";
export const SET_CORPORATE_SIGNATURE = "SET_CORPORATE_SIGNATURE";
export const SET_PERSONAL_SIGNATURE = "SET_PERSONAL_SIGNATURE";
export const SET_DISCLAIMER = "SET_DISCLAIMER";
export const SET_DRAFT_COLLAPSED_CONTENT = "SET_DRAFT_COLLAPSED_CONTENT";
export const SET_DRAFT_EDITOR_CONTENT = "SET_DRAFT_EDITOR_CONTENT";
export const SET_MAX_MESSAGE_SIZE = "SET_MAX_MESSAGE_SIZE";
export const SET_SAVED_INLINE_IMAGES = "SET_SAVED_INLINE_IMAGES";
export const SET_TEMPLATE_CHOOSER_TARGET = "SET_TEMPLATE_CHOOSER_TARGET";
export const SET_TEMPLATE_CHOOSER_VISIBLE = "SET_TEMPLATE_CHOOSER_VISIBLE";
export const SET_TEMPLATE_LIST_LOADING = "SET_TEMPLATE_LIST_LOADING";
export const SET_TEMPLATE_LIST_SEARCH_PATTERN = "SET_TEMPLATE_LIST_SEARCH_PATTERN";
export const SET_TEMPLATES_LIST = "SET_TEMPLATES_LIST";
export const SHOW_SENDER = "SHOW_SENDER";
export const UNSET_CORPORATE_SIGNATURE = "UNSET_CORPORATE_SIGNATURE";

// selection
export const UNSELECT_CONVERSATION = "UNSELECT_CONVERSATION";
export const SELECT_CONVERSATION = "SELECT_CONVERSATION";
export const SET_SELECTION = "SET_SELECTION";
export const UNSELECT_ALL_CONVERSATIONS = "UNSELECT_ALL_CONVERSATIONS";

// consultPanel
export const SET_BLOCK_REMOTE_IMAGES = "SET_BLOCK_REMOTE_IMAGES";
export const SET_CURRENT_EVENT = "SET_CURRENT_EVENT";
export const SET_CURRENT_EVENT_STATUS = "SET_CURRENT_EVENT_STATUS";

// folderList
export const RESET_FOLDER_FILTER_LIMITS = "RESET_FOLDER_FILTER_LIMITS";
export const SET_FOLDER_FILTER_PATTERN = "SET_FOLDER_FILTER_PATTERN";
export const SET_FOLDER_FILTER_RESULTS = "SET_FOLDER_FILTER_RESULTS";
export const SET_FOLDER_FILTER_LIMIT = "SET_FOLDER_FILTER_LIMIT";
export const SET_FOLDER_FILTER_LOADED = "SET_FOLDER_FILTER_LOADED";
export const SET_FOLDER_FILTER_LOADING = "SET_FOLDER_FILTER_LOADING";
export const TOGGLE_EDIT_FOLDER = "TOGGLE_EDIT_FOLDER";

// route
export const SET_ROUTE_FILTER = "SET_ROUTE_FILTER";
export const SET_ROUTE_FOLDER = "SET_ROUTE_FOLDER";
export const SET_ROUTE_MAILBOX = "SET_ROUTE_MAILBOX";
export const SET_ROUTE_SEARCH = "SET_ROUTE_SEARCH";

// conversations
export const ADD_CONVERSATIONS = "ADD_CONVERSATIONS";
export const REMOVE_CONVERSATIONS = "REMOVE_CONVERSATIONS";
export const RESET_CONVERSATIONS = "RESET_CONVERSATIONS";
export const SET_CONVERSATIONS_LOADING_STATUS = "SET_CONVERSATIONS_LOADING_STATUS";
export const SET_CURRENT_CONVERSATION = "SET_CURRENT_CONVERSATION";
export const UNSET_CURRENT_CONVERSATION = "UNSET_CURRENT_CONVERSATION";

// settings
export const SET_MAIL_THREAD_SETTING = "SET_MAIL_THREAD_SETTING";

// preview
export const SET_PREVIEW_PART_ADDRESS = "SET_PREVIEW_PART_ADDRESS";
export const SET_PREVIEW_MESSAGE_KEY = "SET_PREVIEW_MESSAGE_KEY";
