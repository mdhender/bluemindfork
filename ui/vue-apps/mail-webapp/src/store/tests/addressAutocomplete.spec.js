import Vue from "vue";
import Vuex from "vuex";
import cloneDeep from "lodash.clonedeep";
import { ADDRESS_AUTOCOMPLETE_LIST } from "~/getters";
import { ADD_ADDRESS_WEIGHT } from "~/mutations";
import storeData from "../addressAutocomplete";

Vue.use(Vuex);

describe("address autocomplete", () => {
    let store;

    beforeEach(() => {
        store = new Vuex.Store(cloneDeep(storeData));
    });

    test("ADD_ADDRESS_WEIGHT mutation & ADDRESS_AUTOCOMPLETE_LIST getter", () => {
        const address1 = "address-1";
        expect(store.state.addressWeights[address1]).toBeFalsy();
        store.commit(ADD_ADDRESS_WEIGHT, { address: address1, weight: 1 });
        expect(store.state.addressWeights[address1]).toEqual(1);
        store.commit(ADD_ADDRESS_WEIGHT, { address: address1, weight: 1 });
        expect(store.state.addressWeights[address1]).toEqual(2);
        store.commit(ADD_ADDRESS_WEIGHT, { address: address1, weight: -1 });
        expect(store.state.addressWeights[address1]).toEqual(1);
        store.commit(ADD_ADDRESS_WEIGHT, { address: address1, weight: 1 });

        const address2 = "address-2";
        store.commit(ADD_ADDRESS_WEIGHT, { address: address2, weight: 3 });

        const address3 = "address-3";
        store.commit(ADD_ADDRESS_WEIGHT, { address: address3, weight: -1 });

        expect(store.state.addressWeights).toMatchSnapshot();

        const sortedAddresses = store.getters[ADDRESS_AUTOCOMPLETE_LIST];
        expect(sortedAddresses[0]).toEqual(address2);
        expect(sortedAddresses[1]).toEqual(address1);
        expect(sortedAddresses[2]).toEqual(address3);
    });
});
