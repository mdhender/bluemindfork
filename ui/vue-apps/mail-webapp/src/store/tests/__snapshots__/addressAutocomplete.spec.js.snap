// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`address autocomplete ADD_ADDRESS_WEIGHT mutation & ADDRESS_AUTOCOMPLETE_LIST getter 1`] = `
Object {
  "address-1": 2,
  "address-2": 3,
  "address-3": -1,
}
`;
